utils/s2n_mem.c

```diff
@@ -252,6 +264,9 @@ int s2n_free_object(uint8_t **p_data, uint32_t size)
     return s2n_free(&b);
 }
 
+/**
+ * Copy the data in `from` to a new allocated blob.
+ */
```

Comment:
This one always trips me up. I wonder if we should be even more specific about the fact that `to` will be allocated as part of this operation. "Allocate enough memory for `to` to contain all the data in `from`, then copy the data in `from` to `to`.